### PR TITLE
Add test dual

### PR DESCRIPTION
**Pull request by tzufgoogle**
Added validation on the test set for the best model found in validaiton. The metrics output is also changed to the test set

### CODE REVIEW

File: cabby/model/text/dual_encoder/train.py [hunk 1]
```diff
@@ -185,17 +194,35 @@ def train_model(self):
...
+    util.save_metrics_last_only(
+          self.metrics_path, 
           true_points, 
           pred_points)
-        self.save_cell_embed()
 
-        self.model.train()
+    self.save_cell_embed()
 
-    logging.info('Finished Training.')
```
Comment (sayalikulkarni) [inline cabby/model/text/dual_encoder/train.py:198]: Too many blank lines at the end of the file

File: cabby/model/text/dual_encoder/train.py [hunk 2]
```diff
@@ -185,17 +194,35 @@ def train_model(self):
...
+
+    model_state = util.load_checkpoint(self.model_path, self.model, self.device)
+    valid_loss = model_state['valid_loss']
+
+    logging.info(
+      f'Loaded best model (with validation loss {valid_loss}) for testing.')
+    
+    logging.info('Start testing.')
+
+    test_loss, predictions, true_vals, true_points, pred_points = self.evaluate(
```
Comment (sayalikulkarni) [inline cabby/model/text/dual_encoder/train.py:212]: suggestion: since you have the true and predicted points here for every step, we can compute the distance metrics along with accuracy in the same loop. That way you wont need to run the evaluation code separately.
